packages/blade/src/components/StepGroup/_decisions/decisions.md

```diff
@@ -0,0 +1,228 @@
+| color     | color of icon   | 'positive' \| 'negative' \| 'neutral' \| 'notice' \| 'information' \| 'primary' | 'neutral'         |
+
+### StepItemIndicator
+
+| **Props** | **Description**    | **Type**                                                                        | **Default Value** |
+| --------- | ------------------ | ------------------------------------------------------------------------------- | ----------------- |
+| color     | color of indicator | 'positive' \| 'negative' \| 'neutral' \| 'notice' \| 'information' \| 'primary' | 'neutral'         |
+
+## Usage
+
+### Static StepGroup
+
+Static StepGroup does not have any hover, focus, and active states. It can't be selected or clicked. It is for static steps only.
+
+<img width="200px" src="./2024-04-12-14-12-50.png" alt=""/>
+
+```jsx
+<StepGroup>
+  <StepItem
+    leading={<StepItemIndicator color="positive" />}
+    title="Disputes Raised"
+    timestamp="Thu, 11th Oct23 | 12:00pm"
+    stepProgress="full"
+  />
+  <StepItem
+    leading={<StepItemIndicator color="notice" />}
+    title="Needs Response"
+    timestamp="Respond latest by Tue, 23rd Oct'24 | 12:00pm"
+    description="Description"
+    stepProgress="start"
+  >
+    <Button variant="secondary">Submit Documents</Button>
+  </StepItem>
+  <StepItem
+    leading={<StepItemIndicator color="neutral" />}
+    title="Decision from Bank"
+    trailing={<Badge color="neutral">Pending</Badge>}
+  />
+</StepGroup>
+```
+
+### Interactive StepGroup
+
+Interactive StepGroup can be used to add clickable items in StepGroup. When StepItem has `onClick` or `href` prop, it becomes interactive.
+
+<img width="200px" src="./2024-04-12-15-16-25.png" alt="" />
+
+```jsx
+const InteractiveStepGroup = () => {
+  const [selectedIndex, setSelectedIndex] = React.useState(-1);
+
+  return (
+    <StepGroup>
+      <StepItem
+        leading={<StepItemIcon icon={CheckIcon} color="positive" />}
+        title="Introduction"
+        timestamp="Thu, 11th Oct23 | 12:00pm"
+        isSelected={selectedIndex === 0}
+        // `onClick` prop turns item into Interactive item
+        onClick={() => {
+          setSelectedIndex(0);
+        }}
+      />
+      <StepItem
+        leading={<StepItemIcon icon={ClockIcon} color="primary" />}
+        title="Compliance Details"
+        description="Provide documentation of reports"
+        isSelected={selectedIndex === 1}
+        onClick={() => {
+          setSelectedIndex(1);
+        }}
+      />
+    </StepGroup>
+  );
+};
+```
+
+### Nested StepGroup
+
+<img width="200px" src="./2024-04-12-15-22-54.png" alt="" />
+
+```jsx
+<StepGroup>
+  <StepItem
+    leading={<StepItemIcon icon={CheckIcon} color="positive" />}
+    title="Disputes Raised"
+    timestamp="Thu, 11th Oct23 | 12:00pm"
+    stepProgress="full"
+    trailing={<Badge>Label</Badge>}
+  />
+  {/* Nested StepGroup */}
+  <StepGroup>
+    <StepItem
```

Comment:
So we need to follow Nested `StepGroup` pattern to show indentation. Is it possible to handle indentation via a prop? Something like:

```js
<StepGroup>
  <StepItem />
  <StepItem />
  <StepItem parentIndex="1" />
  <StepItem parentIndex="1" />
  <StepItem />
</StepGroup>
```

Reply:
yup. But we follow this structure for nesting in List component and common structure for nesting in compound components.

Reply:
Makes sense then.